src/HOHQMesh.jl

```diff
@@ -32,8 +36,22 @@ function generate_mesh(control_file;
 
   # Determine output filenames
   filebase = splitext(basename(control_file))[1]
-  if isnothing(mesh_filename)
-    mesh_filename = filebase * ".mesh"
+
+  # Find the file line that gives the mesh file format
+  file_lines = readlines(open(control_file))
+  file_idx = findfirst(contains("mesh file format"), file_lines)
+
+  # Extract the mesh file format keyword
+  mesh_file_format = split(file_lines[file_idx])[5]
+
+  if mesh_file_format == "ISM" || mesh_file_format == "ISM-V2" || mesh_file_format == "ISM-v2"
+    if isnothing(mesh_filename)
+      mesh_filename = filebase * ".mesh"
+    end
+  elseif mesh_file_format == "ABAQUS"
+    if isnothing(mesh_filename)
+      mesh_filename = filebase * ".inp"
+    end
```

Comment:
```suggestion
  if isnothing(mesh_filename)
    # Find the file line that gives the mesh file format
    file_lines = readlines(open(control_file))
    line_index = findfirst(contains("mesh file format"), file_lines)

    # Extract the mesh file format keyword
    mesh_file_format = split(file_lines[line_index])[5]

    if mesh_file_format == "ISM" || mesh_file_format == "ISM-V2" || mesh_file_format == "ISM-v2"
      mesh_filename = filebase * ".mesh"
    elseif mesh_file_format == "ABAQUS"
      mesh_filename = filebase * ".inp"
    else
      error("unknown mesh file format: ", mesh_file_format, " (must be one of ISM, ISM-V2, or ABAQUS)")
```
If would keep the check `isnothing(mesh_filename)` in the beginning, since if the filename is specified, there's nothing to do here. Other than that, you could also consider moving this auxiliary format extraction logic into its own function (`extract_mesh_file_format`?) such that the main code remains a little easier to read, but I don't think it's a crucial change.

Reply:
This is a good idea. But I foresee that having such an error message would decrease coverage (probably down to 92% or so) because I am not sure how to make a test that is designed to fail to exercise this part of the code.

Reply:
However, I am in favor of such a message because it gives the user more information about what went wrong.

Reply:
You could create a dummy file, only add a line that specifies a bad mesh file format, and then call `generate_mesh` on it. Something along the lines of
```julia
julia> mktemp() do path, io
         write(io, "mesh file format = ABBAKISS")
         flush(io)
         println(read(path, String))
       end
mesh file format = ABBAKISS
```
where you would replace the `println(...)` line with `generate_mesh(path)` (or with the newly created `extract_mesh_file_format` function)

Reply:
I like the name `ABBAKISS` :). This is exactly what I wanted to do but didn't know how.

Reply:
My implementation of this suggestion records an error rather than a pass. How do I tell Julia that error means a pass in this instance?

Reply:
Something like
```julia
@test_throws ErrorException generate_mesh(path)
```
should do the trick

Reply:
yup that did it. Thanks! I will cleanup and push this